.github/workflows/coding-standards.yml

```diff
@@ -0,0 +1,101 @@
+name: PHP Coding Standards
+
+on:
+    push:
+        branches:
+            - trunk
+            - 'release/**'
+            - 'wp/**'
+    pull_request:
+        paths:
+            # Any change to a PHP file should run checks.
+            - '**.php'
+            # These files configure Composer. Changes could affect the outcome.
+            - 'composer.*'
+            # This file configures PHPCS. Changes could affect the outcome.
+            - 'phpcs.xml.dist'
+            # Changes to workflow files should always verify all workflows are successful.
+            - '.github/workflows/*.yml'
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    # Runs PHP coding standards checks.
+    #
+    # Violations are reported inline with annotations.
+    #
+    # Performs the following steps:
+    # - Checks out the repository.
+    # - Sets up PHP.
+    # - Configures caching for PHPCS scans.
+    # - Installs Composer dependencies.
+    # - Make Composer packages available globally.
+    # - Runs PHPCS on the full codebase with warnings suppressed.
+    # - Generate a report for displaying issues as pull request annotations.
+    # - Runs PHPCS on the `tests` directory without warnings suppressed.
+    # - Generate a report for displaying `test` directory issues as pull request annotations.
+    # - Ensures version-controlled files are not modified or deleted.
+    phpcs:
+        name: PHP coding standards
+        runs-on: ubuntu-latest
+        timeout-minutes: 20
+        if: ${{ github.repository == 'WordPress/gutenberg' || github.event_name == 'pull_request' }}
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+
+            - name: Set up PHP
+              uses: shivammathur/setup-php@1a18b2267f80291a81ca1d33e7c851fe09e7dfc4 # v2.22.0
+              with:
+                  php-version: '7.4'
+                  coverage: none
+                  tools: cs2pr
+
+            # This date is used to ensure that the PHPCS cache is cleared at least once every week.
+            # http://man7.org/linux/man-pages/man1/date.1.html
+            - name: "Get last Monday's date"
+              id: get-date
+              run: echo "date=$(/bin/date -u --date='last Mon' "+%F")" >> $GITHUB_OUTPUT
```

Comment:
What have you measured as the speedup in the CI jobs system?

Reply:
In my tests,
```
phpcs -n --report-full 
```
takes about 12 seconds when the cache is disabled and 0 seconds (that's what GitHub shows) when there is a cache hit.

Reply:
hm. what if we introduced this first without the cache? in the grander scheme of all the test suites 12s is a drop in the bucket but the cache is going to likely create a non-trivial amount of headache at the point it causes problems in a PR.

I don't think these tests are going to be the slowest workflow suite, which means they aren't in the hot path, and if we merge first without the cache then if we think one is truly warranted then we can make a PR and gather a statistical measurement to determine if, when put into the system with 1700 open branches and lots of CI activity, if it still makes 12s of an improvement and if we think that's worth the additional complexity

Reply:
> Can we get an explanation of why we want to clear the cache once a week?
> What is the reason we can't use a content-based cache?

Without having looked at the implementation in this particular workflow...
* Caches in GitHub Actions work differently than people often expect from other CI systems.
* A cache in GHA is created based on a key and NEVER updated.
* The only way to vacate the cache is to create a new cache with a new key.

So, for caches where the key doesn't change often, the cache tends to go more and more out of date over time. Adding a time based factor to the key helps to vacate the cache regularly.

Also see: https://github.com/ramsey/composer-install/issues/234

> hm. what if we introduced this first without the cache?

In case it would be decided to go that way, I would **_strongly_** recommend adding the `--no-cache` CLI argument to the PHPCS command in GHA.
As the `cache` directive is in the ruleset, PHPCS would otherwise needlessly create the cache file from scratch each time, which has a significant impact on the runtime of PHPCS due to the file-write operations.
As CLI args overrule the ruleset, adding `no-cache` should already speed up the CS run.